tests/e2e/isbservice.go

```diff
@@ -363,37 +368,43 @@ func UpdateISBServiceRollout(
+				}
 			}
 
 			Document("Verify that the pipelines are unpaused by checking the PPND conditions on ISBService Rollout and PipelineRollout")
 			Eventually(func() bool {
 				isbRollout, _ := isbServiceRolloutClient.Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
 				isbCondStatus := getRolloutConditionStatus(isbRollout.Status.Conditions, apiv1.ConditionPausingPipelines)
-				plRollout, _ := pipelineRolloutClient.Get(ctx, pipelineRolloutName, metav1.GetOptions{})
-				plCondStatus := getRolloutConditionStatus(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
-				if isbCondStatus == metav1.ConditionTrue || plCondStatus == metav1.ConditionTrue {
-					return false
+				for _, name := range pipelineRolloutNames {
+					plRollout, _ := pipelineRolloutClient.Get(ctx, name, metav1.GetOptions{})
+					plCondStatus := getRolloutConditionStatus(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
+					if isbCondStatus == metav1.ConditionTrue || plCondStatus == metav1.ConditionTrue {
+						return false
+					}
 				}
 				return true
 			}, TestTimeout).Should(BeTrue())
 
 		} else {
 			Document("Verify that dependent Pipeline is not paused when an update to ISBService not requiring pause is made")
 			verifyNotPausing := func() bool {
-				_, _, retrievedPipelineStatus, err := GetPipelineSpecAndStatus(Namespace, pipelineRolloutName)
-				Expect(err).ShouldNot(HaveOccurred())
-				Expect(retrievedPipelineStatus.Phase != numaflowv1.PipelinePhasePaused).To(BeTrue())
 				isbRollout, _ := isbServiceRolloutClient.Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
 				isbCondStatus := getRolloutConditionStatus(isbRollout.Status.Conditions, apiv1.ConditionPausingPipelines)
-				plRollout, _ := pipelineRolloutClient.Get(ctx, pipelineRolloutName, metav1.GetOptions{})
-				plCondStatus := getRolloutConditionStatus(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
-				if isbCondStatus == metav1.ConditionTrue || plCondStatus == metav1.ConditionTrue {
-					return false
-				}
-				if isbRollout.Status.UpgradeInProgress != apiv1.UpgradeStrategyNoOp || plRollout.Status.UpgradeInProgress != apiv1.UpgradeStrategyNoOp {
-					return false
+				for _, name := range pipelineRolloutNames {
+					_, _, retrievedPipelineStatus, err := GetPipelineSpecAndStatus(Namespace, name)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(retrievedPipelineStatus.Phase != numaflowv1.PipelinePhasePaused).To(BeTrue())
+					plRollout, _ := pipelineRolloutClient.Get(ctx, name, metav1.GetOptions{})
+					plCondStatus := getRolloutConditionStatus(plRollout.Status.Conditions, apiv1.ConditionPipelinePausingOrPaused)
+					if isbCondStatus == metav1.ConditionTrue || plCondStatus == metav1.ConditionTrue {
+						return false
+					}
+					if isbRollout.Status.UpgradeInProgress != apiv1.UpgradeStrategyNoOp || plRollout.Status.UpgradeInProgress != apiv1.UpgradeStrategyNoOp {
+						return false
+					}
```

Comment:
I guess this is not a new issue, but it looks like this block is pretty similar to the one above. In fact, the things that this one checks should theoretically be true for the one above, like `isbRollout.Status.UpgradeInProgress` and `retrievedPipelineStatus.Phase != numaflowv1.PipelinePhasePaused`. Can we do some consolidation?